scipy/interpolate/tests/test_bsplines.py

```diff
@@ -430,6 +430,32 @@ def test_neg_axis(self):
         spl1 = BSpline(t, c[1], k)
         assert_equal(spl(2.5), [spl0(2.5), spl1(2.5)])
 
+    @pytest.mark.parametrize('bc_type', ['natural', 'clamped', 'periodic'])
+    def test_from_cubic(self, bc_type):
+        np.random.seed(1234)
+        x = np.sort(np.random.random(20))
+        y = np.random.random(20)
+        if bc_type == 'periodic':
+            y[-1] = y[0]
+        cb = CubicSpline(x, y, bc_type=bc_type)
+        bspl = BSpline.from_cubic(cb, bc_type=bc_type)
+        bspl_new = make_interp_spline(x, y, bc_type=bc_type)
+        assert_allclose(bspl.c, bspl_new.c)
```

Comment:
How close is it? Consider adding explicit atol / rtol.
Also, I'd check that cb(xx), bspl(xx) are close at xx being e.g. halfway between `x` entries

Reply:
Also need at least a smoke test for complex-valued coefficients.